airbyte-config/config-models/src/main/resources/types/WellKnownTypes.yaml

```diff
@@ -0,0 +1,51 @@
+# Regexes are purely illustrative and need to be workshopped: BC dates shouldn't require 4-digit years; years may have >=5 digits; etc
+definitions:
+  String:
+    type: string
+    description: Arbitrary text
+  BinaryData:
+    type: string
+    description: >
+      Arbitrary binary data. Represented as base64-encoded strings in the JSON transport.
+      In the future, if we support other transports, may be encoded differently.
+    # All credit to https://stackoverflow.com/a/475217 for this pattern
+    pattern: (?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?
+  Date:
+    type: string
+    # format: date is a superset of what we want, so we cannot use it here (e.g. it accepts 2-digit years)
+    pattern: \d{4}-\d{2}-\d{2}( BC)?
+  TimestampWithTimezone:
+    type: string
+    # format: date-time is a superset of what we want, so we cannot use it here (e.g. it accepts 2-digit years)
+    pattern: \d{4}-\d{2}-\d{2}[T ]\d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+-]\d{1,2}(:\d{2})?)( BC)?
+    description: An instant in time. Frequently simply referred to as just a timestamp, or timestamptz.
+  TimestampWithoutTimezone:
+    type: string
+    pattern: \d{4}-\d{2}-\d{2}[T ]\d{2}:\d{2}:\d{2}(\.\d+)?( BC)?
+    description: Also known as a localdatetime, or just datetime.
+  TimeWithTimezone:
+    type: string
+    pattern: \d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+-]\d{1,2}(:\d{2})?)
+  TimeWithoutTimezone:
+    type: string
+    pattern: \d{2}:\d{2}:\d{2}(\.\d+)?
+  Number:
+    type: string
+    # Note the mix of regex validation for normal numbers, and enum validation for special values
```

Comment:
This could be a `description `

Reply:
I was planning to remove this comment and have it be a PR comment actually, since it's just commentary about the typedefs rather than actually explaining the types themselves - do you think it's worth leaving in?

Reply:
I was thinking that the regular expressions, examples, and comments in this file are meant to be as complete as possible, as this file will be copied into lots of places (e.g. CDK) and loose any external context.